public/app/features/alerting/unified/components/rule-editor/labels/LabelsFieldOpsLabels.test.tsx

```diff
@@ -0,0 +1,137 @@
+      <SubFormProviderWrapper labels={labels}>
+        <LabelsWithSuggestions dataSourceName="grafana" />
+      </SubFormProviderWrapper>
+    );
+
+    // Wait for the dropdowns to be rendered
+    await waitFor(() => {
+      expect(screen.getAllByTestId('alertlabel-key-picker')).toHaveLength(labels.length);
+    });
+
+    return view;
+  }
+
+  it('should display existing ops labels correctly', async () => {
+    await renderLabelsWithOpsLabels();
+
+    // Verify the keys are displayed
+    expect(screen.getByTestId('labelsInSubform-key-0').querySelector('input')).toHaveValue('sentMail');
+    expect(screen.getByTestId('labelsInSubform-key-1').querySelector('input')).toHaveValue('stage');
+
+    // Verify the values are displayed
+    expect(screen.getByTestId('labelsInSubform-value-0').querySelector('input')).toHaveValue('true');
+    expect(screen.getByTestId('labelsInSubform-value-1').querySelector('input')).toHaveValue('production');
+  });
+
+  it('should render value dropdowns for each label', async () => {
+    await renderLabelsWithOpsLabels();
+
+    // Verify we have value pickers for each label
+    expect(screen.getAllByTestId('alertlabel-value-picker')).toHaveLength(2);
+  });
+
+  it('should allow deleting a label', async () => {
+    const { user } = await renderLabelsWithOpsLabels();
+
+    expect(screen.getAllByTestId('alertlabel-key-picker')).toHaveLength(2);
+
+    await user.click(screen.getByTestId('delete-label-1'));
+
+    expect(screen.getAllByTestId('alertlabel-key-picker')).toHaveLength(1);
+    expect(screen.getByTestId('labelsInSubform-key-0').querySelector('input')).toHaveValue('sentMail');
+  });
+
+  it('should allow adding a new label', async () => {
+    const { user } = await renderLabelsWithOpsLabels();
+
+    await waitFor(() => expect(screen.getByText('Add more')).toBeVisible());
+    await user.click(screen.getByText('Add more'));
+
+    expect(screen.getAllByTestId('alertlabel-key-picker')).toHaveLength(3);
+    expect(screen.getByTestId('labelsInSubform-key-2').querySelector('input')).toHaveValue('');
+  });
+
+  it('should allow typing custom values in dropdowns', async () => {
+    const { user } = await renderLabelsWithOpsLabels();
+
+    // Add a new label
+    await waitFor(() => expect(screen.getByText('Add more')).toBeVisible());
+    await user.click(screen.getByText('Add more'));
+
+    // Type a custom key and value
+    const newKeyInput = screen.getByTestId('labelsInSubform-key-2').querySelector('input');
+    const newValueInput = screen.getByTestId('labelsInSubform-value-2').querySelector('input');
+
+    await user.type(newKeyInput!, 'customKey{enter}');
+    await user.type(newValueInput!, 'customValue{enter}');
+
+    await waitFor(() => {
+      expect(screen.getByTestId('labelsInSubform-key-2').querySelector('input')).toHaveValue('customKey');
+    });
+    expect(screen.getByTestId('labelsInSubform-value-2').querySelector('input')).toHaveValue('customValue');
+  });
```

Comment:
Nice work!
To make sure we don’t regress on the escalation case, could you add a test that explicitly checks:
1. Updating an existing label value still populates the dropdown with correct values
2. The dropdown suggestions are correct for different label keys